cuda_core/cuda/core/experimental/_program.py

```diff
@@ -3,9 +3,361 @@
 # SPDX-License-Identifier: LicenseRef-NVIDIA-SOFTWARE-LICENSE
 
 from cuda import nvrtc
-from cuda.core.experimental._utils import handle_return
+from cuda.core.experimental._utils import handle_return, _handle_boolean_option, check_or_create_options
 from cuda.core.experimental._module import ObjectCode
 
+from typing import Optional, Tuple, Union
+from dataclasses import dataclass
+
+@dataclass
+class ProgramOptions:
+    """Customizable :obj:`ProgramOptions` for NVRTC.
+
+    Attributes
+    ----------
```

Comment:
Q: have you checked that the Sphinx doc builds fine with this change? (Note that `ProgramOptions` is new so you need to add it to `api.rst`.)